open-hackathon-server/src/hackathon/hack/host_server_manager.py

```diff
@@ -50,6 +53,22 @@ class DockerHostManager(Component):
     hosted_docker = RequiredFeature("hosted_docker")
     sche = RequiredFeature("scheduler")
 
+    def get_all_docker_hosts(self, hackathon_id):
+        """
+        Get all host servers of a hackathon
+        :param hackathon_id: the id of this hackathon
+        :type hackathon_id: integer
+
+        :return: a list of all docker hosts
+        :rtype: list
+        """
+        vms = self.db.find_all_objects(DockerHostServer, DockerHostServer.hackathon_id == hackathon_id)
+        def get_data(vm):
+            data = vm.dic()
+            return data
+
+        return map(lambda vm: get_data(vm), vms)
```

Comment:
inner function `get_data` does simple things. You can simply the lines into one: `return [vm.dic() for vm in vms]`